### ISSUE TITLE
weights for datasets

### ISSUE THREAD

**Issue by niklasnolte** [zfit/zfit#193]
Hi,
I am trying to perform a fit on a weighted dataset, first trying the simple example in https://github.com/zfit/zfit/blob/develop/examples/simple_fit.py.   
To do that, I used the `weights` argument in `zfit.Data.from_numpy`.  
My weights are `[0] * 9999 + [1]`. When fitting, it tells me that there are incompatible shapes:
```
  File "iminuit/_libiminuit.pyx", line 736, in iminuit._libiminuit.Minuit.migrad
RuntimeError: exception was raised in user function
User function arguments:
       mu = +1.200000
    sigma = +1.300000
Original python exception in user function:
InvalidArgumentError: Incompatible shapes: [10000] vs. [9992]
	 [[node mul (defined at lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py:1748) ]]
```
Are the weights meant to be used for something else? Or is this unwanted behaviour?

**Comment by mayou36** [comment]
Hi, the weights are supposed to be event weights, that is true. But there could indeed be a problem with data ranges applied. Is it possible that some of your events are not within the range or your `Space` and only 9992 are inside? (so 8 outside?).

It's a bug then for sure, it's unwanted behavior.

**Comment by niklasnolte** [comment]
yeah, increasing the range did it.  
i guess wanted behaviour would be to cut the weights in the same way as the data when fitting a specified space

**Comment by mayou36** [comment]
> yeah, increasing the range did it.
> i guess wanted behaviour would be to cut the weights in the same way as the data when fitting a specified space

yes, exactly, as this is currently under full refactoring, this will soon be implemented correctly.

Since yours works now (and in general, cutting the data before should be the preferred way I think) and it will simply fail (so no unnoticed bug), I'll leave the issue open to close it when we have things refactored.

Do you agree on that? Let me know if you have more troubles with it or it's a blocker, then we'll tackle it ASAP

**Comment by niklasnolte** [comment]
i just do the cutting myself for now. thanks for taking care of it!

**Comment by mayou36** [comment]
Fixed in #212 (currently develop). Thanks again for reporting and please let us know in case the problem still persists.
